Azure/sonic-buildimage
# [Services] Allow monit system tool to monitor the critical processes status running in various SONiC containers.

- What I did
I added a new configuration file called monit_**_container_**  (container at here represent the container
name) for each container and then copied this config file into the base image under the directory
/etc/monit/config.d. In the host, monit will detect the status of all critical processes running in 
various docker containers. Specifically, monit will detect whether a process is in the dead or live 
state. If a process which is monitored went into the dead state or it failed to start, then monit will
write an alert into syslog. Currently monit will periodically monitor all the process in the period
120 seconds. We can set a different value for this period.

- How I did it
I created this configuration file according to the template config file named monitrc in the host.

- How to verify it
Before we test this, we should disable the auto-restart feature since if we kill a critical process, 
the corresponding container will restart and then the processes will come up again.
In the host, we can list the critical processes running in each docker container using
the command docker top container_id. After that, we deliberately kill a process
using the command kill -9 proc_id. Then we can review the syslog file and will see an
alert message showing process_name is not running.
